tests/core/tools/array.js

```diff
@@ -151,8 +151,35 @@
 				assert.areSame( 0, index );
 				arrayAssert.itemsAreSame( [ 1234 ], array );
 			} );
-		}
+		},
+
+		'test array.find': function() {
+			var arr = [ 'foo', 'bar', 'baz', 1, 2, 3 ],
+				results = [],
+				ret;
+
+			ret = this.array.find( arr, function( item, index ) {
+				results.push( item );
+
+				assert.areSame( arr.indexOf( item ), index, 'Index argument should match item index.' );
+
+				return false;
+			} );
 
+			assert.isUndefined( ret, 'Undefined should be returned.' );
```

Comment:
`Undefined should be returned` it is not a good practice to put expected value if you use proper assertion.

In this case you used the proper assertion, `assert.isUndefined()` which already tells exactly what value is expected. If assertion fails, there is an information that `undefined` was expected. And this should be **the only** source of truth.

Duplicating this information in assertion msg you add a risk that someone will change the method, but will leave msg unchanged.

Simply make it value agnostic, but keep the information what has failed, like: "Proper value is returned" or even simpler as we do in many other places "Returned value".